### DISCUSSION TITLE
SharpTools Integration

### DISCUSSION THREAD

**Discussion by Pdub76** [RobertD502/home-assistant-flair#73] [General]
Afternoon Robert and other contributors

I'm extremely grateful for the HA integration, and I noticed that the only way to control the vent position is with the slider.  For me jut the toggle open/close produces no results.
I'm also building out my dashboard with SharpTools and hoping someone here can assist me with some html to set/change the cover tilt position.
Thanks in advance!

**Comment by RobertD502** [comment]
The vent has 3 positions possible: 0 (closed), 50 ,100 (open). The slider is mainly there to allow users to set the vent position to 50% open.

When you say "toggle open/close", are you referring to these two buttons in the UI?:
![image](https://github.com/RobertD502/home-assistant-flair/assets/52541649/e03c7bbf-e703-4766-8af5-064f4551a6c9)

Or are you saying you are trying to open and close a vent/vents with a service call?


Regarding SharpTools, I won't be able to be of any assistance since I don't use it, but there may be some other users out there - I'd ask for help in the SharpTools community forums.

**Reply by RobertD502** [reply]
If you are trying to toggle between open and closed and are trying to do this via a service call then you need to be using the service `cover.toggle_cover_tilt`

**Reply by Pdub76** [reply]
Hi Robert, first off, thanks for the quick response.

Yes, in the UI if I tap the arrow it doesn't toggle the open / close control.
However, if I change the control to the slider, it does trigger the 50%, open or close action.
![image](https://github.com/RobertD502/home-assistant-flair/assets/145679486/7ddb9e91-2184-4ec1-9cee-cd16ac514f32)

**Reply by RobertD502** [reply]
> Hi Robert, first off, thanks for the quick response.
> 
> Yes, in the UI if I tap the arrow it doesn't toggle the open / close control.
> However, if I change the control to the slider, it does trigger the 50%, open or close action.
> ![image](https://github.com/RobertD502/home-assistant-flair/assets/145679486/7ddb9e91-2184-4ec1-9cee-cd16ac514f32)
> 
> 

Should work as expected since both setting the slider to 0 and pressing the close arrow call the same API endpoint (and slider to 100 and open arrow).

One thing is, there is a delay on Flair's side - when you call open/close on a vent, the Flair servers send the command to the puck, but it can take a bit of time (up to several minutes at times) until the puck actually sends the command to the vent.

**Reply by Pdub76** [reply]
That's what I thought as well with the programming.  Either option, slider if arrows, would trigger the even.  However, in my case inky the slider works regardless of the time I wait (1 min or 10 mins).  Lol 
Oh well, guess I'll just leave it on an auto mode and use my hubs to monitor and display the status and temperatures in each room.  No worries, just thought I'd  see if anyone else had encountered the same issues or not.

Thanks again Robert, very appreciative of the integration and functionality!

Paul

**Reply by Pdub76** [reply]
Okay, tech is making a liar of me.
Just retested the arrow toggles in HA and they triggered the open close toggle as expected.
Also rebuilt the sharptools custom hub tile and configured it with the same toggle command and it triggered the open close as well.

Hope you have a great weekend!

Paul
